common/src/chain/transaction/signature/inputsig/arbitrary_message/mod.rs

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) 2021-2024 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const MESSAGE_MAGIC_PREFIX: &str = "===MINTLAYER MESSAGE BEGIN===";
+const MESSAGE_MAGIC_SUFFIX: &str = "===MINTLAYER MESSAGE END===";
+
+use thiserror::Error;
+
+use serialization::Encode;
+
+use crate::{
+    chain::{signature::DestinationSigError, ChainConfig, Destination},
+    primitives::{id::default_hash, H256},
+};
+
+use super::{
+    authorize_pubkey_spend::{
+        sign_pubkey_spending, verify_public_key_spending, AuthorizedPublicKeySpend,
+    },
+    authorize_pubkeyhash_spend::{
+        sign_address_spending, verify_address_spending, AuthorizedPublicKeyHashSpend,
+    },
+    classical_multisig::authorize_classical_multisig::{
+        verify_classical_multisig_spending, AuthorizedClassicalMultisigSpend,
+    },
+};
+
+#[derive(Debug, Error, PartialEq, Eq)]
+pub enum SignArbitraryMessageError {
+    #[error("Destination signature error: {0}")]
+    DestinationSigError(#[from] DestinationSigError),
+    #[error("AnyoneCanSpend should not use standard signatures, so producing a signature for it is not possible")]
+    AttemptedToProduceSignatureForAnyoneCanSpend,
+    #[error("Classical multisig signature attempted in uni-party function")]
+    AttemptedToProduceClassicalMultisigSignatureInUnipartySignatureCode,
+    #[error("Unsupported yet!")]
+    Unsupported,
+}
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct SignedArbitraryMessage {
+    raw_signature: Vec<u8>,
```

Comment:
Seems to me `ArbitraryMessageSignature` would be a more appropriate name. Seems the struct does not actually contain the message.

Reply:
Done.